File: python_project/encapsulamiento/punto.py
# Punto
# Crear la clase Punto que tendrá dos atributos numéricos, x e y, que 
# representarán las coordenadas del punto dentro del plano. Implemente los siguientes métodos:
# estaSobreElEjeX(): devuelve valor booleano
# estaSobreElEjeY(): devuelve valor boolean
# esElOrigenDeCoordenadas(): devuelve valor booleano
# distanciaAlOrigen(): devuelve valor numérico
# distancia(Punto p1, Punto p2): devuelve valor numérico
# distancia(Punto p): devuelve valor numérico

class Punto:
    def __init__(self, x : float, y : float) -> None:
        self.x = x
        self.y = y

    def esta_sobre_eje_x(self) -> bool:
        
        return self.y == 0
        
    def esta_sobre_eje_y(self) -> bool:
        
        return self.x == 0 

    def es_origen(self) -> bool:

        return self.x == 0 and self.y == 0

    def distancia_al_origen(self) -> float: 

        return (self.x**2 + self.y**2)**(1/2)

    def distancia(self, otro: 'Punto') -> float:

        return ((self.x - otro.x)**2 + (self.y - otro.y)**2)**(1/2)

    @staticmethod
    def distancia2(p1: 'Punto', p2: 'Punto') -> float:
        
        return ((p1.x - p2.x)**2 + (p1.y - p2.y)**2)**0.5

    def __str__(self) -> str:
        
        return(f'[{self.x} ; {self.y}]')

if __name__ == "__main__":
    p1 = Punto(3, 4)
    p2 = Punto(0, 2)

    print(f'La distancia al origen es: {p1.distancia_al_origen()} u.')
    print(f'La distancia al origen es: {p2.distancia_al_origen()} u.')

    print(f'Esta sobre el eje x? {p1.esta_sobre_eje_x()}')
    print(f'Esta sobre el eje x? {p2.esta_sobre_eje_x()}')

    print(f'Esta sobre el eje x? {p1.esta_sobre_eje_y()}')
    print(f'Esta sobre el eje x? {p2.esta_sobre_eje_y()}')

    print(f'La distancia entre {p1} y {p2} es de {p1.distancia(p2)} u.')

    print(Punto.distancia2(p1, p2))

File: java_project/encapsulamiento/puntos/Punto.java
package encapsulamiento.puntos;

// Punto
// Crear la clase Punto que tendrá dos atributos numéricos, x e y, que 
// representarán las coordenadas del punto dentro del plano. Implemente los siguientes métodos:
// estaSobreElEjeX(): devuelve valor booleano
// estaSobreElEjeY(): devuelve valor boolean
// esElOrigenDeCoordenadas(): devuelve valor booleano
// distanciaAlOrigen(): devuelve valor numérico
// distancia(Punto p1, Punto p2): devuelve valor numérico
// distancia(Punto p): devuelve valor numérico

public class Punto {

    private double x;
    private double y;


    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Punto() {
    }

    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }

    public boolean estaSobreElEjeX(){

        return this.y == 0; 
    }

    public boolean estaSobreElEjeY(){

        return this.x == 0;
    }

    public boolean esElOrigenDeCoordenadas(){

        return this.x == 0 && this.y == 0;
    }

    public double distanciaAlOrigen(){

        return Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2));
    }

    public static Double distancia(Punto p1, Punto p2){

        return Math.sqrt(Math.pow((p1.x - p2.x), 2) + Math.pow((p1.y - p2.y), 2));
    }

    public double distancia(Punto otro){

        return Math.sqrt(Math.pow((this.x - otro.x), 2) + Math.pow((this.y - otro.y), 2));
    }

    @Override
    public String toString() {
        return "Punto [x = " + getX() + ", y = " + getY() + "]";
    }
    
}

// package encapsulamiento.puntos;

// // Punto
// // Crear la clase Punto que tendrá dos atributos numéricos, x e y, que 
// // representarán las coordenadas del punto dentro del plano. Implemente los siguientes métodos:
// // estaSobreElEjeX(): devuelve valor booleano
// // estaSobreElEjeY(): devuelve valor boolean
// // esElOrigenDeCoordenadas(): devuelve valor booleano
// // distanciaAlOrigen(): devuelve valor numérico
// // distancia(Punto p1, Punto p2): devuelve valor numérico
// // distancia(Punto p): devuelve valor numérico

// public class Punto {
//     // primero declarar atributos.
//     private float x;
//     private float y;
//     // segundo el constructor.

//     public Punto(float x, float y) {
//         this.x = x;
//         this.y = y;
//     }

//     public float getX() {
//         return x;
//     }

//     public void setX(float newX) {
//         x = newX;
//     }

//     public float getY() {
//         return y;
//     }

//     public void setY(float y) {
//         this.y = y;
//     }

//     public boolean estaSobreElEjeX() {
//         return this.y == 0;
//     }

//     public boolean estaSobreElEjeY() {
//         return this.x == 0;
//     }

//     public boolean esElOrigenDeCoordenadas() {
//         return this.x == 0 && this.y == 0;
//     }

//     public double distanciaAlOrigen() {
//         // √((x)^2 + (y)^2)
//         double xx = Math.pow(x, 2.0); // o multiplicar las veces necesarias xd
//         double yy = Math.pow(y, 2.0);

//         return Math.sqrt(xx + yy);
//     }

//     public static float distancia(Punto p1, Punto p2) {
//         // √((x1 - x2)^2 + (y1 - y2)^2)
//         float xx = (float) Math.pow((p1.x - p2.x), 2.0);
//         float yy = (float) Math.pow((p1.y - p2.y), 2.0);

//         return (float) Math.sqrt(xx + yy);
//     }

//     public float distancia(Punto p) {
//         float xx = (float) Math.pow((this.x - p.x), 2.0);
//         float yy = (float) Math.pow((this.y - p.y), 2.0);

//         return (float) Math.sqrt(xx + yy);
//     }

//     @Override
//     public String toString() {
//         return String.format("%n==============PUNTO==============%n" +
//                 "%nPunto en x: " + this.x +
//                 "%nPunto en y: " + this.y +
//                 "%nEsta sobre el eje X?: " + estaSobreElEjeX() +
//                 "%nEsta sobre el eje Y?: " + estaSobreElEjeY() +
//                 "%nEsta en el origen: " + esElOrigenDeCoordenadas() +
//                 "%nLa distancia al origen es: " + distanciaAlOrigen());

//     }
// }
